marshal.go

```diff
@@ -224,31 +214,37 @@ func (l *SchemaList) extractIfPresent(m map[string]interface{}, name string) err
 // a single `map[string]interface{}` to initialize this list
 // of schemas
 func (l *SchemaList) Extract(v interface{}) error {
-	switch v.(type) {
+	switch val := v.(type) {
 	case []interface{}:
-		src := v.([]interface{})
-		*l = make([]*Schema, len(src))
+		*l = make([]*Schema, len(val))
 		var s *Schema
-		for i, d := range src {
-			if err := extractSingleSchema(&s, d); err != nil {
+		for i, d := range val {
+			m, ok := d.(map[string]interface{})
+			if !ok {
+				return errors.Wrap(
+					errInvalidType("map[string]interface{}", d),
+					"failed to extract schema list",
+				)
+			}
+			if err := extractSingleSchema(&s, m); err != nil {
 				return errors.Wrap(err, "failed to extract schema list")
 			}
 			(*l)[i] = s
 		}
 		return nil
 	case map[string]interface{}:
 		var s *Schema
-		if err := extractSingleSchema(&s, v); err != nil {
+		if err := extractSingleSchema(&s, val); err != nil {
 			return errors.Wrap(err, "failed to extract schema list")
 		}
 		*l = []*Schema{s}
 		return nil
+	default:
+		return errors.Wrap(
```

Comment:
I moved a couple of these returns into default statements - that way if someone forgets the return statement at the end of the case, the compiler will complain, rather than silently falling through to the error case at the end of the function.